CONTRIBUTING.md

```diff
@@ -0,0 +1,35 @@
+# Contributing to Chariot 
+
+## Introduction
+
+This document serves as the reference for developers who would like to get the development environment setup and understand the intended workflow required to contribute to this project. This is still a work in progress as the project evolves.
+
+## Setting up the dev environment
+
+These instructions are split into those for the official [front-end](#setting-up-the-front-end) and the [core](#setting-up-the-core), since the core works independent of the front-end and exposes a REST-ful API which enables any developer to write their own front-end.
+
+### Setting up the front end
+
+WIP
+
+### Setting up the core
+
+The entirety of the core is written in Python. Thus the following are required in order to have a working dev environment:
+
+- NIX* environment (Linux, WSL, macOS terminal)
+- Python (3.5 and above)
+- Pip
+
+Keep in mind that we enforce the use of python3 type hints as much as possible, to avoid obscurity.
+
+Optionally, though strongly recommended, is to install Python's virtual environment (venv) module which will help with package management and dependency isolation. The rest of this tutorial assumes you have chosen to use a venv. 
+
+Follow these steps to finish setting up:
+1. Spin up a venv in the core folder, and activate it using any equivalent shell command to bash's `source ./bin/activate`.
```

Comment:
This step isn't clear to me.

Reply:
Right, thanks for the catch. Will add instructions on how to start up a venv.

Reply:
@ryanhassing take another look.

Reply:
'spin up a venv in the core folder' - still not obvious to me what this means. Maybe just write the CLI steps needed:
`cd core`
`venv`
or whatever steps are necessary.

Reply:
As the tag on this code block says, that is the outdated version that you first read. Comment on the changes from the latest commit.